eox_core/edxapp_wrapper/backends/enrollment_h_v1.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Backend for the create_edxapp_user that works under the open-release/hawthorn.beta1 tag
+"""
+
+from __future__ import absolute_import, unicode_literals
+from enrollment import api
+from enrollment.errors import CourseModeNotFoundError
+from student.models import (
+    AlreadyEnrolledError,
+    CourseFullError,
+    EnrollmentClosedError,
+    NonExistentCourseError
+)
+from course_modes.models import CourseMode
+
+
+def create_enrollment(*args, **kwargs):
+
+    errors = []
+    user_id = kwargs.get('username')
+    course_id = kwargs.get('course_id')
+    mode = kwargs.get('mode')
+    is_active = kwargs.get('is_active', True)
+    enrollment_attributes = kwargs.get('enrollment_attributes', None)
+    validation_errors = check_edxapp_enrollment_is_valid(*args, **kwargs)
+    if validation_errors:
+        return None, [", ".join(validation_errors)]
+    try:
+        enrollment = api._data_api().create_course_enrollment(user_id, course_id, mode, is_active)
+        if enrollment_attributes is not None:
+            api.set_enrollment_attributes(user_id, course_id, enrollment_attributes)
+    except (NonExistentCourseError, EnrollmentClosedError, CourseFullError, AlreadyEnrolledError) as e:
+        enrollment = None
+        errors.append(repr(e))
+    return enrollment, errors
+
+
+def check_edxapp_enrollment_is_valid(*args, **kwargs):
+    errors = []
+    is_active = kwargs.get("force_registration", False)
```

Comment:
likely an error is_active <=> force_registration

Reply:
or just duplicated and we can leave the one on L#45

Reply:
done (removed duplicate)